realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1951,6 +1949,21 @@ private OsResults lazyFindAll() {
         return this;
     }
 
+    /**
+     * Limits the number of objects returned in case the query matched more objects.
+     *
+     * @param limit a limit that is {@code &gt; 0}.
```

Comment:
Is limit(0) allowed?

Reply:
I think technically yes, it would just evaluate to a FALSE predicate.